tests/unit/s2n_security_policy_cert_preferences_test.c

```diff
@@ -156,6 +160,20 @@ int main(int argc, char **argv)
         DEFER_CLEANUP(struct s2n_cert_chain_and_key *invalid_cert = NULL, s2n_cert_chain_and_key_ptr_free);
         EXPECT_SUCCESS(s2n_test_cert_permutation_load_server_chain(&invalid_cert, "rsae", "pss", "4096", "sha384"));
 
+        struct s2n_security_policy rfc9151_applied_locally = security_policy_rfc9151;
+        rfc9151_applied_locally.certificate_preferences_apply_locally = true;
+
+        /* s2n_connection_set_cipher_preferences looks up the security policy from the security_policy_selection table
+         * but none of our current security policies apply certificate preferences locally. So instead we rewrite the
+         * rfc9151 policy from the table to apply cert preference locally. */
+        for (int i = 0; security_policy_selection[i].version != NULL; i++) {
+            if (strcasecmp("rfc9151", security_policy_selection[i].version) == 0) {
+                security_policy_selection[i].security_policy = &rfc9151_applied_locally;
+                break;
```

Comment:
You need to undo this change once you're tested with it. You shouldn't be changing this for every other test that follows. The scope of where this change is active needs to be very clear.

You had to clean up a bunch of old unit tests where the interactions between tests caused problems :)